src/storage/object.ts

```diff
@@ -352,10 +331,16 @@ export class ObjectStorage {
     await this.db.testPermission((db) => {
       return Promise.all([
         db.findObject(this.bucketId, sourceObjectName, 'id'),
-        db.createObject({
+        db.updateObject(this.bucketId, sourceObjectName, {
+          name: sourceObjectName,
+          version: '1',
+          owner,
+        }),
+        db.asSuperUser().createObject({
```

Comment:
why is this testpermission check required?

Reply:
probably better to have it as a separate check? or add a comment.. clubbing it in testpermission isnt obvious cos I was thinking about what permissions is this clause checking for (esp when it is being run as a superuser?